src/crimson/os/seastore/seastore_types.h

```diff
@@ -36,29 +36,139 @@ struct seastore_meta_t {
 };
 
 // Identifies segment location on disk, see SegmentManager,
-using segment_id_t = uint32_t;
+using internal_segment_id_t = uint32_t;
```

Comment:
I see internal_segment_id_t and device_segment_id_t used almost interchangeably in this PR.  What does each one mean?  Where is each meant to be used?

Reply:
I added internal_segment_id_t to represent a segment id without device id, I thought this might be useful when we are referencing segment ids within the segment manager scope. Am I right?

Reply:
Ok, what does device_segment_id_t mean?

Reply:
segment ids with a device id encapsulated, it's used in segment_id_t. I think I see your point, maybe I should hide device_segment_id_t from outside segment_id_t, am I right?

Reply:
Right, only one of those two types should be visible outside of segment_id_t.  I'd also like you to swap them.  internal_segment_id_it should be "internal" to segment_id_t and not visible outside of it (you can define the type in the private: section of segment_id_t).  Let device_segment_id_t be the one used externally.

Reply:
Got it:-)